irc/irc.py

```diff
@@ -20,75 +20,43 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-import socket
-import sys
 import logging
 import base64
+import pydle
 from config import Config
-from concurrent import futures
 
-class IRC:
-    def __init__(self, agent):
+class IRC(pydle.Client):
+    async def on_connect(self):
+        await self.join(self.channel)
+        logging.info("Bot Join %s" % (self.channel))
+
+    async def on_message(self, target, source, message):
+        if source == Config.conf()["admin"]:
+            if message.startswith("CreateGame ") == True:
+                self.CreateGame(message)
```

Comment:
Note: These might be blocking, so maybe we might have to refactor IRC.CreateGame() to be an async function by moving the blocking gRPC call to a thread pool, and have that thread pool invoke callbacks on the asyncio message queue.

However, such a refactor can be costly, so maybe we'll do it when we have time.